--- a/units/console-getty.service.in
+++ b/units/console-getty.service.in
@@ -22,7 +22,7 @@
 [Service]
 # The '-o' option value tells agetty to replace 'login' arguments with an option to preserve environment (-p),
 # followed by '--' for safety, and then the entered username.
-ExecStart=-/sbin/agetty -o '-p -- \\u' --noclear --keep-baud - 115200,38400,9600 $TERM
+ExecStart=-/usr/sbin/agetty --noclear --keep-baud console 115200,57600,38400,9600 $TERM
 Type=idle
 Restart=always
 UtmpIdentifier=cons
--- a/units/container-getty@.service.in
+++ b/units/container-getty@.service.in
@@ -27,7 +27,7 @@
 [Service]
 # The '-o' option value tells agetty to replace 'login' arguments with an option to preserve environment (-p),
 # followed by '--' for safety, and then the entered username.
-ExecStart=-/sbin/agetty -o '-p -- \\u' --noclear - $TERM
+ExecStart=-/usr/sbin/agetty --noclear --keep-baud pts/%I 115200,57600,38400,9600 $TERM
 Type=idle
 Restart=always
 RestartSec=0
--- a/units/getty@.service.in
+++ b/units/getty@.service.in
@@ -36,9 +36,9 @@
 [Service]
 # the VT is cleared by TTYVTDisallocate
 # The '-o' option value tells agetty to replace 'login' arguments with an
 # option to preserve environment (-p), followed by '--' for safety, and then
 # the entered username.
-ExecStart=-/sbin/agetty -o '-p -- \\u' --noclear - $TERM
+ExecStart=-/usr/sbin/agetty --noclear %I $TERM
 Type=idle
 Restart=always
 RestartSec=0
--- a/units/serial-getty@.service.in
+++ b/units/serial-getty@.service.in
@@ -32,8 +32,8 @@ 
 [Service]
 # The '-o' option value tells agetty to replace 'login' arguments with an
 # option to preserve environment (-p), followed by '--' for safety, and then
 # the entered username.
-ExecStart=-/sbin/agetty -o '-p -- \\u' --keep-baud 115200,57600,38400,9600 - $TERM
+ExecStart=-/usr/sbin/agetty ttyAMA0 --login-program /bin/ash --login-options "--login" --skip-login --keep-baud 115200,57600,38400,9600 %I $TERM
 Type=idle
 Restart=always
 UtmpIdentifier=%I
--- a/units/serial-getty@.service.in
+++ b/units/serial-getty@.service.in
@@ -60,2 +60,3 @@
 [Install]
 WantedBy=getty.target
+DefaultInstance=ttyAMA0
