--- a/arch/arm/boot/dts/st/Makefile
+++ b/arch/arm/boot/dts/st/Makefile
@@ -65,7 +65,25 @@ dtb-$(CONFIG_ARCH_STM32) += \
 	stm32mp157c-lxa-tac-gen2.dtb \
 	stm32mp157c-odyssey.dtb \
 	stm32mp157c-osd32mp1-red.dtb \
-	stm32mp157c-phycore-stm32mp1-3.dtb
+	stm32mp157c-phycore-stm32mp1-3.dtb \
+	stm32mp1-seeed-lcd-01-overlay.dtbo \
+	stm32mp1-seeed-pwm-overlay.dtbo \
+	stm32mp1-seeed-spi5-overlay.dtbo \
+	stm32mp1-seeed-uart7-overlay.dtbo \
+	stm32mp1-seeed-usart2-overlay.dtbo \
+	stm32mp1-seeed-i2c4-overlay.dtbo \
+	stm32mp1-seeed-ap6236-overlay.dtbo \
+	stm32mp1-MCP2517FD-can0-overlay.dtbo
+
+DTC_FLAGS_stm32mp157c-odyssey += -@
+DTC_FLAGS_stm32mp1-seeed-lcd-01-overlay += -@
+DTC_FLAGS_stm32mp1-seeed-pwm-overlay += -@
+DTC_FLAGS_stm32mp1-seeed-spi5-overlay += -@
+DTC_FLAGS_stm32mp1-seeed-uart7-overlay += -@
+DTC_FLAGS_stm32mp1-seeed-usart2-overlay += -@
+DTC_FLAGS_stm32mp1-seeed-i2c4-overlay += -@
+DTC_FLAGS_stm32mp1-seeed-ap6236-overlay += -@
+DTC_FLAGS_stm32mp1-MCP2517FD-can0-overlay += -@
 dtb-$(CONFIG_ARCH_U8500) += \
 	ste-snowball.dtb \
 	ste-hrefprev60-stuib.dtb \
