diff --git a/drivers/gpio/gpio-davinci.c b/drivers/gpio/gpio-davinci.c
index 76b58c70b257..f2cc94c260cd 100644
--- a/drivers/gpio/gpio-davinci.c
+++ b/drivers/gpio/gpio-davinci.c
@@ -62,6 +62,7 @@ struct davinci_gpio_controller {
 	int			irqs[MAX_INT_PER_BANK];
 	struct davinci_gpio_regs context[MAX_REGS_BANKS];
 	u32			binten_context;
+	bool		needs_context_restore;
 };
 
 static inline u32 __gpio_mask(unsigned gpio)
@@ -79,6 +80,7 @@ static inline struct davinci_gpio_regs __iomem *irq2regs(struct irq_data *d)
 }
 
 static int davinci_gpio_irq_setup(struct platform_device *pdev);
+int davinci_gpio_resume_all_devices(void);
 
 /*--------------------------------------------------------------------------*/
 
@@ -221,6 +223,7 @@ static int davinci_gpio_probe(struct platform_device *pdev)
 	chips->chip.request = gpiochip_generic_request;
 	chips->chip.free = gpiochip_generic_free;
 #endif
+	chips->needs_context_restore = false;
 	spin_lock_init(&chips->lock);
 
 	chips->gpio_unbanked = gpio_unbanked;
@@ -632,6 +635,7 @@ static int davinci_gpio_suspend(struct device *dev)
 	u32 nbank = DIV_ROUND_UP(chips->chip.ngpio, 32);
 
 	davinci_gpio_save_context(chips, nbank);
+	chips->needs_context_restore = true;
 
 	return 0;
 }
@@ -641,7 +645,10 @@ static int davinci_gpio_resume(struct device *dev)
 	struct davinci_gpio_controller *chips = dev_get_drvdata(dev);
 	u32 nbank = DIV_ROUND_UP(chips->chip.ngpio, 32);
 
-	davinci_gpio_restore_context(chips, nbank);
+	if (chips->needs_context_restore) {
+		davinci_gpio_restore_context(chips, nbank);
+		chips->needs_context_restore = false;
+	}
 
 	return 0;
 }
@@ -666,6 +673,18 @@ static struct platform_driver davinci_gpio_driver = {
 	},
 };
 
+static int davinci_gpio_resume_wrapper(struct device *dev, void *unused)
+{
+	return davinci_gpio_resume(dev);
+}
+
+int davinci_gpio_resume_all_devices(void)
+{
+	return driver_for_each_device(&davinci_gpio_driver.driver, NULL,
+					NULL, davinci_gpio_resume_wrapper);
+}
+EXPORT_SYMBOL(davinci_gpio_resume_all_devices);
+
 /*
  * GPIO driver registration needs to be done before machine_init functions
  * access GPIO. Hence davinci_gpio_drv_reg() is a postcore_initcall.
